### DISCUSSION TITLE
HTML to Groff (ms) links not shown

### DISCUSSION THREAD

**Discussion by TheCarnacki** [jgm/pandoc#10291] [Q&A] [answered]
I'm having difficulty getting links to appear when I convert from HTML to Groff (ms). Here's a simple HTML test case:

```
<!DOCTYPE html>
<html>
<head>
        <title>Test Page</title>
        <style type="text/css">
                a {
                        color: green;
                }
        </style>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph. Here's a <a href="foo.html">link</a>!</p>

</body>
</html>
```

and the resulting Groff code:

```
.SH 1
My First Heading
.pdfhref O 1 "My First Heading"
.pdfhref M "my-first-heading"
.LP
My first paragraph.
Here\[aq]s a \c
.pdfhref W -D "foo.html" -A "\c" \
 -- "link"
\&!
```

When I convert using groff -ms all the content appears except for the linked text. Pdfroff gives me a similar result. I must be missing some simple step. Any ideas? Thank you.

**Comment by jgm** [comment, marked as answer]
Use `pdfroff` with `-ms -mpdfmark -mspdf`

**Comment by TheCarnacki** [comment]
That did it. Thank you!

**Reply by jgm** [reply]
General hint:  use `pandoc -t ms -s -o output.pdf input.html --verbose` and you'll see the pdfroff options pandoc itself will use when creating a pdf via ms.
